### PR TITLE
[feature/meal] meal/get 핸들러 구현

### PR DESCRIPTION
**Pull request by tpdns90321**
1. NetworkError의 상태코드가 404에서 502(BadGateway)로 변경
2. InvaildError 에러코드 추가, 상태코드는 400(BadRequest)
3. meal/get 구현 완료

### CODE REVIEW

**Comment by sukso96100** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 9
           

Complexity increasing per file
==============================
- meal.go  6
- error.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160&bid=19944202)

File: meal.go [hunk 1]
```diff
@@ -93,21 +94,25 @@ func GetMealIds(c *gin.Context) {
 	})
 }
 
+// 실제 식단 데이터
```
Comment (sukso96100) [inline meal.go:97]: Codacy found an issue: [comment on exported type Diet should be of the form "Diet ..." (with optional leading article)](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

File: meal.go [hunk 2]
```diff
@@ -93,21 +94,25 @@ func GetMealIds(c *gin.Context) {
...
 type Diet struct {
 	Diet    string `json:"diet"`
 	Calorie string `json:"calorie"`
 }
 
+// 점심 학식
```
Comment (sukso96100) [inline meal.go:103]: Codacy found an issue: [comment on exported type Lunch should be of the form "Lunch ..." (with optional leading article)](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

File: meal.go [hunk 3]
```diff
@@ -175,3 +178,144 @@ func getMealDataFromID(client HttpClient, id int) (week []MealData, err error) {
...
+// @Param id query int false "게시물 ID" 382
+// @Param day query int false "날짜 요일 (0~5)" 5
+// @Success 200 {object} GetMealDataResult
+// @Failure 400 {object} ErrorMessage
+// @Failure 404 {object} ErrorMessage
+// @Failure 502 {object} ErrorMessage
+// @Router /meal/get [get]
+func GetMealData(c *gin.Context) {
+	// GET 인수를 받는다. 기본값은 문자열 "0"
+	plainId := c.DefaultQuery("id", "0")
```
Comment (sukso96100) [inline meal.go:277]: Codacy found an issue: [var plainId should be plainID](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

File: meal.go [hunk 4]
```diff
@@ -93,21 +94,25 @@ func GetMealIds(c *gin.Context) {
...
 }
 
+// 점심 학식
 type Lunch struct {
 	A Diet `json:"a"`
 	B Diet `json:"b"`
 	C Diet `json:"c"`
 }
 
+// 저녁 학식
```
Comment (sukso96100) [inline meal.go:110]: Codacy found an issue: [comment on exported type Dinner should be of the form "Dinner ..." (with optional leading article)](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

File: meal.go [hunk 5]
```diff
@@ -175,3 +178,144 @@ func getMealDataFromID(client HttpClient, id int) (week []MealData, err error) {
...
+			week, err = getMealDataWithID(client, id.ID)
+			if err != nil {
+				return
+			}
+
+			return
+		}
+	}
+
+	err = NotFoundError.CreateError(errors.New("해당 기간에 학식이 존재하지 않습니다."))
```
Comment (sukso96100) [inline meal.go:236]: Codacy found an issue: [error strings should not be capitalized or end with punctuation or a newline](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

File: meal.go [hunk 6]
```diff
@@ -175,3 +178,144 @@ func getMealDataFromID(client HttpClient, id int) (week []MealData, err error) {
...
+		data[weekday-1],
+	}
+	return
+}
+
+type GetMealDataResult struct {
+	Data []MealData `json:"data"`
+}
+
+// godoc GetMealData
```
Comment (sukso96100) [inline meal.go:264]: Codacy found an issue: [comment on exported function GetMealData should be of the form "GetMealData ..."](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

File: meal.go [hunk 7]
```diff
@@ -93,21 +94,25 @@ func GetMealIds(c *gin.Context) {
...
 	B Diet `json:"b"`
 	C Diet `json:"c"`
 }
 
+// 저녁 학식
 type Dinner struct {
 	A Diet `json:"a"`
 }
 
+// 학식 데이터
```
Comment (sukso96100) [inline meal.go:115]: Codacy found an issue: [comment on exported type MealData should be of the form "MealData ..." (with optional leading article)](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

File: meal.go [hunk 8]
```diff
@@ -175,3 +178,144 @@ func getMealDataFromID(client HttpClient, id int) (week []MealData, err error) {
...
+
+var regexDateFromTitle *regexp.Regexp = regexp.MustCompile("(\\d{1,2})\\D{1,2}(\\d{1,2})\\D{1,2}(\\d{1,2})\\D{1,2}(\\d{1,2})\\D?")
+
+// 현재 시간을 기준으로 해당 주의 시간표를 추출한다.
+func getMealDataWithTime(client HttpClient, now time.Time) (week []MealData, err error) {
+	defer WhereInError(&err, "현재 주간 학식")
+
+	weekday := now.Weekday()
+	if weekday < time.Monday || weekday > time.Friday {
+		err = NotFoundError.CreateError(errors.New("주말에는 학식이 제공하지 않습니다."))
```
Comment (sukso96100) [inline meal.go:190]: Codacy found an issue: [error strings should not be capitalized or end with punctuation or a newline](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

File: meal.go [hunk 9]
```diff
@@ -175,3 +178,144 @@ func getMealDataFromID(client HttpClient, id int) (week []MealData, err error) {
...
+		return
+	}
+
+	day = []MealData{
+		data[weekday-1],
+	}
+	return
+}
+
+type GetMealDataResult struct {
```
Comment (sukso96100) [inline meal.go:260]: Codacy found an issue: [exported type GetMealDataResult should have comment or be unexported](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

Comment (sukso96100) [reply meal.go:260]: 이거 문서화 뺴먹은듯

**Comment by sukso96100** [comment]
문서화 빼먹은거랑 변수 네이밍 자동 리뷰 붙은것만 좀 고쳐봐

**Comment by sukso96100** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
- Added 2
           

Complexity increasing per file
==============================
- meal.go  6
- error.go  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160&bid=19944202)

File: meal.go [hunk 10]
```diff
@@ -159,19 +162,162 @@ func getMealDataFromID(client HttpClient, id int) (week []MealData, err error) {
...
+			week, err = getMealDataWithID(client, id.ID)
+			if err != nil {
+				return
+			}
+
+			return
+		}
+	}
+
+	err = NotFoundError.CreateError(errors.New("해당 기간에 학식이 존재하지 않습니다."))
```
Comment (sukso96100) [inline meal.go:237]: Codacy found an issue: [error strings should not be capitalized or end with punctuation or a newline](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

File: meal.go [hunk 11]
```diff
@@ -159,19 +162,162 @@ func getMealDataFromID(client HttpClient, id int) (week []MealData, err error) {
...
+// 제목에서 기간을 추출하기 위한 정규표현식
+var regexDateFromTitle *regexp.Regexp = regexp.MustCompile("(\\d{1,2})\\D{1,2}(\\d{1,2})\\D{1,2}(\\d{1,2})\\D{1,2}(\\d{1,2})\\D?")
+
+// 현재 시간을 기준으로 해당 주의 시간표를 추출한다.
+func getMealDataWithTime(client HttpClient, now time.Time) (week []mealData, err error) {
+	defer WhereInError(&err, "현재 주간 학식")
+
+	weekday := now.Weekday()
+	if weekday < time.Monday || weekday > time.Friday {
+		err = NotFoundError.CreateError(errors.New("주말에는 학식이 제공하지 않습니다."))
```
Comment (sukso96100) [inline meal.go:191]: Codacy found an issue: [error strings should not be capitalized or end with punctuation or a newline](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160)

File: meal.go (hunk 10 above)
Comment (tpdns90321) [reply meal.go:237]: 한국어를 대문자로 인식하냐ㅋㅋㅋㅋ

**Comment by sukso96100** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 5
           

Complexity increasing per file
==============================
- error.go  2
- meal.go  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/s-owl/skhuinfo-api/pullRequest?prid=6091160&bid=19944202)

**Review by sukso96100 — APPROVED** [review]

**Pull request merged**